crates/fj-core/src/services/service.rs

```diff
@@ -62,6 +62,12 @@ impl<S: State> Deref for Service<S> {
     }
 }
 
+impl<S: State> DerefMut for Service<S> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.state
+    }
+}
```

Comment:
I think this is used by the new `Services` constructor, which in itself seems fine. But this provides direct access to service state to *all* code, which seems error-prone.

I think there are two better ways to do this:

1. Define a command for updating the validation config (see the existing `ValidationCommand`), make the execution of that command emit `ValidationEvent::ConfigurationDefined`, and use `Service::execute` to update the validation.
2. In the new `Services` constructor, instead of using `Service::<Validation>::default()`, initialize the validation service with `Service::new`, which can be provided with a `Validation` that has the configuration set properly.

Solution 1 is more verbose, and probably more flexible than we need. So I'd prefer solution 2. What do you think?

Reply:
I think I agree with you, solution 1 seems like it might be overkill, so I will try my hand at solution 2 and see how that goes. :+1:

Reply:
@hannobraun , I believe I have managed to implement all of the changes that you had suggested. Please take a look and let me know if this is what you were going for! :smile: